src/OSPSuite.R/Services/DimensionTask.cs

```diff
@@ -0,0 +1,43 @@
+﻿using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.R.Domain.UnitSystem;
+
+namespace OSPSuite.R.Services
+{
+   public interface IDimensionTask
+   {
+      IDimension DimensionByName(string dimensionName);
+
+      IDimension DimensionForUnit(string unit);
+
+      double[] ConvertToUnit(IDimension dimension, string targetUnit, double[] valuesInBaseUnit, double? molWeight = null);
+
+      double[] ConvertToUnit(string dimensionName, string targetUnit, double[] valuesInBaseUnit, double? molWeight = null);
+   }
+
+   public class DimensionTask : IDimensionTask
+   {
+      private readonly IDimensionFactory _dimensionFactory;
+
+      public DimensionTask(IDimensionFactory dimensionFactory)
+      {
+         _dimensionFactory = dimensionFactory;
+      }
+
+      public IDimension DimensionByName(string dimensionName) => _dimensionFactory.Dimension(dimensionName);
+
+      public IDimension DimensionForUnit(string unit) => _dimensionFactory.DimensionForUnit(unit);
+
+      public double[] ConvertToUnit(IDimension dimension, string targetUnit, double[] valuesInBaseUnit, double? molWeight = null)
```

Comment:
This can probably fail in a 1000 ways. We have to see how it's going to be called in R